### PR TITLE
Drop apiextensions

### PR DESCRIPTION
**Pull request by fiunchinho**
Towards https://github.com/giantswarm/giantswarm/issues/21301

## Checklist

- [x] Update changelog in CHANGELOG.md.
- [ ] If the change introduces new roles, please consider adding user-friendly descriptions to the roles with the annotation `giantswarm.io/notes` to help explain their purpose and usage.

### CODE REVIEW

File: service/controller/clusternamespace/clusternamespace.go [hunk 1]
```diff
@@ -40,14 +39,15 @@ func NewClusterNamespace(config ClusterNamespaceConfig) (*ClusterNamespace, erro
 
 	var clusterNamespaceController *controller.Controller
 	{
-		selector := newSelector(func(labels labels.Labels) bool {
-			return labels.Has(label.Organization) && labels.Has(label.Cluster)
-		})
+		selector, err := labels.Parse(fmt.Sprintf("%s,%s", label.Organization, label.Cluster))
```
[the remark this answers is not in the document]
Comment (tfussell) [reply service/controller/clusternamespace/clusternamespace.go:42]: Appears to be correct to me.

**Review by anvddriesch — APPROVED** [review]

File: service/controller/clusternamespace/resource/clusternamespaceresources/create.go [hunk 2]
```diff
@@ -4,8 +4,8 @@ import (
 	"context"
 	"fmt"
 
-	securityv1alpha1 "github.com/giantswarm/apiextensions/v3/pkg/apis/security/v1alpha1"
 	"github.com/giantswarm/microerror"
+	orgcrd "github.com/giantswarm/organization-operator/api/v1alpha1"
```
Comment (tfussell) [inline service/controller/clusternamespace/resource/clusternamespaceresources/create.go:8]: I would use an alias of `security` as that's the name of the API group for the Organization API.

File: service/controller/rbac/rbac.go [hunk 3]
```diff
@@ -44,14 +43,15 @@ func NewRBAC(config RBACConfig) (*RBAC, error) {
 
 	var namespaceAuthController *controller.Controller
 	{
-		selector := newSelector(func(labels labels.Labels) bool {
-			return labels.Has(label.Organization) || labels.Has(legacylabel.LegacyCustomer)
-		})
+		selector, err := labels.Parse(fmt.Sprintf("%s,%s", label.Organization, legacylabel.LegacyCustomer))
```
Comment (tfussell) [inline service/controller/rbac/rbac.go:46]: The previous selector was OR `||`, but `,` means AND so this changes the logic. And according to https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/, `Caution: For both equality-based and set-based conditions there is no logical OR (||) operator. Ensure your filter statements are structured accordingly.` We'll need to figure out the implication here.

Comment (fiunchinho) [reply service/controller/rbac/rbac.go:46]: Taking a deeper look it seems here we are filtering `Namespaces` that are labeled with the `organization` label or with the `customer` label. On `kvm` installations I found namespaces using the `customer` label which doesn't use `organization` label. So I believe we have two options:
- We remove the `customer` part of the selector and we don't upgrade `rbac-operator` on kvm installations
- We remove the whole selector and add the filter in all resources executed by this controller

What do you think?

Comment (anvddriesch) [reply service/controller/rbac/rbac.go:46]: I think it would be better to remove the selector and add a check to all resources.

Comment (fiunchinho) [reply service/controller/rbac/rbac.go]: Moved selector to individual handlers, please take a look.

Comment (anvddriesch) [reply service/controller/rbac/rbac.go]: If it's okay, can the check be in a function in the rbac package with a comment explaining what it does?

Comment (fiunchinho) [reply service/controller/rbac/rbac.go]: Sure! Done.

Comment (anvddriesch) [reply service/controller/rbac/rbac.go]: Thanks! :D Works for me.

File: service/controller/clusternamespace/resource/clusternamespaceresources/create.go (hunk 2 above)
Comment (fiunchinho) [reply service/controller/clusternamespace/resource/clusternamespaceresources/create.go]: done

**Pull request merged**